dart/dynamics/Entity.h

```diff
@@ -63,11 +63,23 @@ class Shape;
 /// may have different policies about how/if their reference frame or name
 /// can be changed. Use the Detachable class to create an Entity whose reference
 /// Frame can be changed arbitrarily.
-class Entity
+class Entity : public common::Subscription
 {
 public:
   friend class Frame;
 
+  enum Notification {
```

Comment:
There seems to be a drawback with strongly typed enums: The compiler refuses to implicitly cast them to int, even if it is declared with the size of "int". This means that a type cast is needed to pass an enumerated value into the Publisher::sendNotification(int) function, and a typecast will again be needed to compare that int with an enumerated value.

I try to avoid type casting when possible, because it tends to feel dirty to me. I guess the question now is whether the advantages of strongly typed enumerations outweigh the grossness of casting them over and over again.